tests/Collection/CollectionFunctionalTest.php

```diff
@@ -311,6 +311,68 @@ public function testMapReduce()
         $this->assertNotEmpty($result->getCounts());
     }
 
+    /**
+     * @expectedException MongoDB\Driver\Exception\InvalidArgumentException
+     * @expectedExceptionMessage inherited
+     */
+    public function testInsertOneInTransactionWithInheritedWriteConcern()
+    {
+        $this->skipIfTransactionsAreNotSupported();
+
+        $this->createCollection();
+
+        $session = $this->manager->startSession();
+        $session->startTransaction([
+            'readConcern' => new \MongoDB\Driver\ReadConcern("snapshot"),
+            'writeConcern' => new \MongoDB\Driver\WriteConcern(\MongoDB\Driver\WriteConcern::MAJORITY)
+        ]);
+
+        $collection = new Collection(
+            $this->manager,
+            $this->getDatabaseName(),
+            $this->getCollectionName(),
+            [
+                'writeConcern' => new \MongoDB\Driver\WriteConcern(\MongoDB\Driver\WriteConcern::MAJORITY)
+            ]
+        );
+
+        $collection->insertOne(
+            ['test' => 'foo'],
+            ['session' => $session]
+        );
+
+        $this->dropCollection();
+    }
+
+    /**
+     * @expectedException MongoDB\Driver\Exception\InvalidArgumentException
+     * @expectedExceptionMessage option
```

Comment:
We can still use a substring here, but it should probably be something more specific in the final test. For instance, if the final message is `The "readConcern" option cannot be specified within a transaction. Instead, specify it when starting the transaction.` we might just go with `"readConcern" option cannot be specified within a transaction`.